src/goertzel/goertzel.c

```diff
@@ -0,0 +1,60 @@
+/**
+ * @file goertzel.c  Goertzel algorithm
+ *
+ * Copyright (C) 2010 Creytiv.com
+ */
+
+#include <math.h>
+#include <re.h>
+#include <rem_goertzel.h>
+
+
+#define PI 3.14159265358979323846264338327
+
+
+/**
+ * Initialize goertzel state
+ *
+ * @param g     Goertzel state
+ * @param freq  Target frequency
+ * @param srate Sample rate
+ */
+void goertzel_init(struct goertzel *g, double freq, unsigned srate)
+{
```

Comment:
perhaps check the pointer to Goertzel state?

Reply:
This was avoided on purpose for performance reasons. The goertzel state struct is public and typically not dynamically allocated on its own, but part of another structure, and will therefor not likely be NULL.